### PR TITLE
feat: add support for updating multiple __snapshots__ directories from a single coarse grained jest_test target

### PR DESCRIPTION
**Pull request by gregmagolan**
Updated docstring:

```
        snapshots: If True, a `{name}_update_snapshots` binary target is generated that will update the `__snapshots__`
            directory when `bazel run`. This is the equivalent to running `jest -u` or `jest --updateSnapshot` outside of Bazel.

            If the name of the snapshot directory is not the default `__snapshots__` because of a custom snapshot resolver,
            you can specify a custom directory name by setting `snapshots` to the custom directory name instead of True.

            If there are multiple snapshot directories to update from a single `jest_test` target you can specify a list
            of directories instead. For example,

            ```
            jest_test(
                name = "test",
                config = "jest.config.js",
                data = [
                    "link/link.js",
                    "link/link.test.js",
                    "greetings/greetings.js",
                    "greetings/greetings.test.js",
                ],
                snapshots = [
                    "greetings/__snapshots__",
                    "link/__snapshots__",
                ],
            )
            ```

            Snapshots directories must not contain any files except for snapshots. There must also be no BUILD files in
            the snapshots directories since they must be part of the same Bazel package that the `jest_test` target is in.

            If snapshots are _not_ configured to output to a directory that contains only snapshots, you may alternately
            set `snapshots` to a list of snapshot files expected to be generated by this `jest_test` target.
            These must be source files and all snapshots that are generated must be explicitly listed. You may use a
            `glob` such as `glob(["**/*.snap"])` to generate this list, in which case all snapshots must already be on
            disk so they are discovered by `glob`.
```

### CODE REVIEW

File: jest/defs.bzl [hunk 1]
```diff
@@ -133,23 +155,49 @@ def jest_test(
...
+                )
+                fail(msg)
+            if snapshot_label.name.endswith(snapshots_ext):
+                snapshot_files.append(snapshot_label)
+                snapshot_data.append(snapshot_label)
+            else:
+                snapshot_directories.append(snapshot_label)
+                snapshot_data.extend(native.glob(["{}/**".format(snapshot)]))
+
+    elif snapshots != False:
```
Comment (jbedard) [inline jest/defs.bzl:185]: Isn't changing `None` to `False` a breaking change?

Comment (gregmagolan) [reply jest/defs.bzl:185]: I changed the default to False

Comment (gregmagolan) [reply jest/defs.bzl:185]: but I suppose I could allow for both False & None here to mean the same thing

Comment (gregmagolan) [reply jest/defs.bzl]: Fixed

Comment (jbedard) [reply jest/defs.bzl]: If someone override that default with `None` this would make their build fail.

What was the reason for changing it?

Comment (gregmagolan) [reply jest/defs.bzl]: None is not as description as False in the docs IMO. But I changed it to accept both False & None so it won't be breaking.

File: jest/defs.bzl [hunk 2]
```diff
@@ -179,21 +227,19 @@ def jest_test(
...
             **kwargs
         )
 
         _jest_update_snapshots(
             name = name,
             config = config,
             data = data,
-            tags = tags,
-            update_directory = update_directory,
+            tags = tags + ["manual"],  # tagged manual so it is not built unless run
```
Comment (jbedard) [inline jest/defs.bzl:239]: Some of the manual tags were missing before? Does this fix some targets being built when not needed?

**Review by jbedard — APPROVED** [review]

**Pull request merged**